### PR TITLE
Add checks for number of arguments in the transaction

### PR DESCRIPTION
**Pull request by ilblackdragon**
Errors otherwise are unintelligible.

Is there a better way to do this?

### CODE REVIEW

File: src.ts/transaction.ts [hunk 1]
```diff
@@ -27,6 +27,9 @@ export class AccessKey extends Assignable {
 }
 
 export function fullAccessKey(): AccessKey {
+    if (arguments.length != 0) {
+        throw new Error(`Wrong number of arguments ${arguments.length}: expected none`);
```
Comment (vgrichina) [inline src.ts/transaction.ts]: Let's create specific error here https://github.com/nearprotocol/nearlib/blob/master/src.ts/utils/errors.ts

Comment (vgrichina) [reply src.ts/transaction.ts]: I think it's also best to include arguments into error message instead of just length (faster to debug in most cases)

**Comment by vgrichina** [comment]
I wonder if underlying issue is that we don't have proper errors in Borsh serializer. We know all the field types there, so can verify them generally and report field path with wrong type.

**Comment by vgrichina** [comment]
also make sure this doesn't fail in `ci` env https://travis-ci.com/nearprotocol/nearlib/jobs/268369102

**Comment by ilblackdragon** [comment]
You might be right, the errors in borsh should be improved as well.
It's sad that TS actually doesn't have a way to automatically enforce types when used from JS.

I think you have a better idea how to do this, so I'm going to close this.

**Pull request closed without merging**